internal/pkg/manifest/workload.go

```diff
@@ -819,3 +819,14 @@ func placementStringP(p PlacementString) *PlacementString {
 	placement := p
 	return &placement
 }
+
+func (cfg PublishConfig) publishedTopics() []Topic {
+	var pubs []Topic
+	for _, topic := range cfg.Topics {
```

Comment:
nit: since the topic length is known ahead of time we can speed things up and be a bit more mem efficient slightly like this:
```
pubs := make([]Topic, len(cfg.Topics)
for i, topic := range cfg.Topics {
   // if... { } 
   pubs[i] = topic[i]
}
return pubs
```